students/julienjass/Lesson09/mailroom_oo.py

```diff
@@ -0,0 +1,234 @@
+        return self._donors[donor_name.lower()].total_donations
+
+    def get_donor(self, donor_name):
+        try:
+            return self._donors[donor_name.lower()]
+        except KeyError:
+            print('This Donor doesn''t exist in the database')
+
+    @property
+    def export(self):
+        return {donor.name:donor.donations for donor in self._donors.values()}
+
+    def donor_list(self):
+        name_list = [donor.name for donor in self._donors.values()]
+        return '\n'.join(name_list)
+
+    def letters_on_disk(self):
+        for donor in self._donors.values():
+            print('Creating a letter for {:s}'.format(donor.name))
+            with open(donor.filename, 'w') as outfile:
+                outfile.write(donor.letter)
+        print()
+
+    def report(self):
+        sorted_donors = [donor for donor in self._donors.values()]
+        sorted_donors.sort(key=lambda donor: donor.total_donations, reverse=True)
+        report_rows = []
+        for d in sorted_donors:
+            report_rows.append('{:27s} {:>12s} {:^13d} {:>12s}'.format(d.name, '${:,.2f}'.format(d.total_donations), d.count_donations, '${:,.2f}'.format(d.average_donation)))
+        header = ('Donor Name                | Total Given | Num Gifts | Average Gift\n') + ('-' * 66) + '\n'
+        return header + '\n'.join(report_rows) + '\n'
+
+    @property
+    def count_donors(self):
+        return len(self._donors)
+
+    @property
+    def count_donations(self):
+        return sum([len(donor.donations) for donor in self._donors.values()])
+
+    @property
+    def total_donations(self):
+        return sum([donor.total_donations for donor in self._donors.values()])
+
+    @property
+    def average_total_donation(self):
+        return self.total_donations / self.count_donors
+
+    @property
+    def average_single_donation(self):
+        return self.total_donations / self.count_donations
+
+    def __repr__(self):
+        return str([d for d in self._donors.values()])
+
+
+class Menu:
+    
+    response = None
+
+    def __init__(self, title, menu):
+        self._title = title
+        self._menu = menu
+        error_msg = 'This is not a valid response. Please enter '
+        for key in self._menu:
+            error_msg += key + ', '
+        self._error = error_msg[:-3] + 'or ' +  error_msg[-3] + '.\n'
+
+    def menu(self):
+        m = [str(k) + ') ' + str(v[0]) + '\n' for k, v in self._menu.items()]
+        return ''.join(m)
+
+    def get_response(self):
+        print(self._title)
+        print(self.menu())
+        response = input('Your choice: ')
+        while response not in self._menu:
+            print(self._error)
+            response = input('Your choice: ')
+        self.response = response
+
+    @property
+    def switch(self):
+        return {k:v[1] for k, v in self._menu.items()}
+
+    
+def print_report():
+    print(donor_db.report())
+
+
+def return_to_main():
+    return
+
```

Comment:
I'd look for another way to represent this instead of creating a function with nothing but a bare `return`